python/ray/workflow/common.py

```diff
@@ -189,7 +189,7 @@ def to_metadata(self) -> Dict[str, Any]:
         # "name" in the metadata is different from the "name" field.
         # Maybe rename it to avoid confusion.
         metadata = {
-            "name": get_module(f) + "." + get_qualname(f),
+            "name": self.name or get_module(f) + "." + get_qualname(f),
```

Comment:
@iycheng I have also made the change to address https://github.com/ray-project/ray/pull/20156#discussion_r770032815

I think the change is valid, but not sure if we want to include the change in this PR.  Let me know your thought and we can always revert this commit and do it in another place.

Reply:
i'm ok with this.